### DISCUSSION TITLE
reference/docs/lua

### DISCUSSION THREAD

**Discussion by giscus** [jaywcjlove/reference#460] [Q&A]
# reference/docs/lua

https://wangchujiang.com/reference/docs/lua.html

**Comment by CloudIsDeep** [comment]
lua中还有goto语法

goto Label

::Label::

**Reply by jaywcjlove** [reply]
@CloudIsDeep 欢迎提交 PR

**Reply by CloudIsDeep** [reply]
for i = 0, 10, 1 do
    print(i)
    if i == 5 then
        -- Label 为标签名 可自定义
        -- 满足i == 5时将跳到对应的标签位置
        goto Label 
    end
end

:: Label  ::
print('结束')

**Reply by jaywcjlove** [reply]
@CloudIsDeep 已添加

**Reply by CloudIsDeep** [reply]
在lua中return不能这样用哦，如果想让脚本停止你可以写成do return end

**Reply by CloudIsDeep** [reply]
像这样
local function isValidNumber(num)
    if type(num) ~= "number" then
        goto invalidNumber   -- 无条件地跳转到代码中的标签 `::invalidNumber::`
    end

    print("Valid number:", num)
    do return end

    ::invalidNumber::
    print("Invalid number:", num)
end

isValidNumber(123)    -- 输出: Valid number: 123
isValidNumber("abc")  -- 输出: Invalid number: abc

**Reply by jaywcjlove** [reply]
@CloudIsDeep 欢迎提交 PR
